pkg/controllers/provisioning/v1alpha1/reallocator/terminate.go

```diff
@@ -0,0 +1,58 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reallocator
+
+import (
+	"context"
+	"fmt"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+type Terminator struct {
+	kubeClient client.Client
+}
+
+func (t *Terminator) CordonNode(ctx context.Context, node *v1.Node) error {
+	if node.Spec.Unschedulable {
```

Comment:
Is this check still necessary?

Reply:
I haven't added a query to get non-cordoned nodes yet. I was going to do this in a new PR. When I add it in, this won't be necessary.